src/dune_engine/cached_digest.ml

```diff
@@ -94,19 +94,25 @@ let delete_very_recent_entries () =
   let now = get_current_filesystem_time () in
   match Float.compare cache.max_timestamp now with
   | Lt -> ()
-  | Eq | Gt ->
-    Path.Table.filteri_inplace cache.table ~f:(fun ~key:path ~data ->
-        match Float.compare data.stats.mtime now with
-        | Lt -> true
-        | Gt | Eq ->
-          if !Clflags.debug_digests then
+  | Eq | Gt -> (
+    let filter (data : file) =
+      match Float.compare data.stats.mtime now with
+      | Lt -> true
+      | Gt | Eq -> false
+    in
+    match !Clflags.debug_digests with
+    | false -> Path.Table.filter_inplace cache.table ~f:filter
```

Comment:
This change is wrong: we want to filter our regardless of the flag. The conditional was only for printing out diagnostic measurements.

Reply:
Unless I'm missing something obvious, we're still doing the filtering.

Reply:
Oh, sorry, you're right!